src/controllers/stallController.ts

```diff
@@ -244,17 +244,66 @@ async function importStalls(req: Request, res: Response, next: NextFunction) {
     let parseError = '';
     let currRow = 2; // header is on first row
 
+    type stallRow = {
+      name: string;
+      description: string;
+      contactNo: string;
+      unitNo: string;
+      openingHours: string;
+      isFeatured: boolean;
+      hawkerCentreId: number;
+    };
+
+    type transformedStallRow = Omit<stallRow, 'openingHours'> & {
+      openingHours: JSON;
+    };
+
     await new Promise((resolve, _reject) => {
-      const stream = parse({ headers: true })
+      const stream = parse<stallRow, transformedStallRow>({
+        headers: true,
+      })
+        .transform(
+          (data: stallRow): transformedStallRow => {
+            const dataArr: string[] = data.openingHours.split(' ');
+            const days: string[] = [
+              'Monday',
+              'Tuesday',
+              'Wednesday',
+              'Thursday',
+              'Friday',
+              'Saturday',
+              'Sunday',
+            ];
+            const formattedOpeningHours: { [key: string]: object } = {};
+            days.forEach(day => {
+              const isClosed = !dataArr.includes(day);
+              const idx = dataArr.indexOf(day);
```

Comment:
Should revert to `closed` if `startTime` or `endTime` is not specified. So we don't get `openingHours` as `0800 - Wednesday`. Could also throw an error to the user.

Reply:
ok I set it to closed if not specified